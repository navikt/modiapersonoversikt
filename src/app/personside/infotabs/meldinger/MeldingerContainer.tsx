import * as React from 'react';
import { Traad } from '../../../../models/meldinger/meldinger';
import styled from 'styled-components';
import theme, { pxToRem } from '../../../../styles/personOversiktTheme';
import TraadVisningContainer from './traadvisning/TraadVisningContainer';
import RestResourceConsumer from '../../../../rest/consumer/RestResourceConsumer';
import VerktoylinjeContainer from './traadvisning/verktoylinje/VerktoylinjeContainer';
import TraadListe from './traadliste/TraadListe';
import { CenteredLazySpinner } from '../../../../components/LazySpinner';

const meldingerMediaTreshold = pxToRem(900);

const MeldingerArticleStyle = styled.article`
    @media (min-width: ${meldingerMediaTreshold}) {
        display: flex;
        align-items: flex-start;
        > *:last-child {
            margin-left: ${theme.margin.layout};
        }
    }
    > * {
        margin-bottom: ${theme.margin.layout};
    }
    position: relative;
    > *:first-child {
        flex: 30% 1 1;
    }
    > *:last-child {
        position: sticky;
        top: 0;
        flex: 70% 1 1;
    }
`;

function MeldingerContainer() {
    return (
        <article>
            <RestResourceConsumer<Traad[]>
                getResource={restResources => restResources.tr√•derOgMeldinger}
                returnOnPending={<CenteredLazySpinner />}
            >
                {data => (
                    <MeldingerArticleStyle>
                        <TraadListe traader={data} />
                        <div>
                            <VerktoylinjeContainer />
                            <TraadVisningContainer />
                        </div>
                    </MeldingerArticleStyle>
                )}
            </RestResourceConsumer>
        </article>
    );
}

export default MeldingerContainer;
