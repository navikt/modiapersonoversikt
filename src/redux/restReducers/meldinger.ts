import { getMeldinger } from '../../api/meldinger-api';
import { createActionsAndReducer } from './restResource';

const { reducer, action, reload, tilbakestill } = createActionsAndReducer('meldinger');

export function hentMeldinger(fødselsnummer: string) {
    return action(() => getMeldinger(fødselsnummer));
}

export function reloadMeldinger(fødselsnummer: string) {
    return reload(() => getMeldinger(fødselsnummer));
}

export function resetMeldingerResource() {
    return tilbakestill;
}

export default reducer;
