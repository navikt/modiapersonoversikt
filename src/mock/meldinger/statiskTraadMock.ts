import { LestStatus, Meldingstype, Traad } from '../../models/meldinger/meldinger';
import { saksbehandlerTekst } from '../../app/personside/infotabs/meldinger/utils/meldingerUtils';
import { Temagruppe } from '../../models/Temagrupper';

export const statiskTraadMock: Traad = {
    traadId: 'sg838exr',
    meldinger: [
        {
            id: 'x4k72bwd',
            meldingstype: Meldingstype.SAMTALEREFERAT_OPPMOTE,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'nzxo4q', fornavn: 'Sander', etternavn: 'Hansen' }),
            journalfortAv: { ident: 'vlzaiw', fornavn: 'Adrian', etternavn: 'Pettersen' },
            fritekst:
                'Ea laborum accusamus et ex voluptates provident eveniet voluptatem. Sunt et sed eligendi harum exercitationem. Eaque est maiores autem sunt occaecati atque ipsa ea itaque. Dolorum enim quaerat alias ut sit labore pariatur.',
            lestDato: '2019-07-12',
            status: LestStatus.Lest,
            opprettetDato: '2019-07-30',
            journalfortDato: '2019-07-11',
            ferdigstiltDato: '2019-07-07',
            erFerdigstiltUtenSvar: true,
            erDokumentMelding: false
        },
        {
            id: 'q1i6ndyj',
            meldingstype: Meldingstype.SPORSMAL_SKRIFTLIG,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'xmqgb8', fornavn: 'Frida', etternavn: 'Pedersen' }),
            journalfortAv: { ident: 'ly4pps', fornavn: 'Emil', etternavn: 'Nguyen' },
            fritekst:
                'Quo magni eaque et libero minima ipsum amet et dolorem. Sed iusto ratione sit. Voluptate alias qui impedit ut dolorum doloribus qui. Qui beatae rerum minima suscipit.',
            lestDato: '2019-07-14',
            status: LestStatus.IkkeLest,
            opprettetDato: '2019-08-10',
            journalfortDato: '2019-07-21',
            ferdigstiltDato: '2019-07-16',
            erFerdigstiltUtenSvar: true,
            erDokumentMelding: false
        },
        {
            id: 'q57oayxa',
            meldingstype: Meldingstype.SPORSMAL_SKRIFTLIG,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'e5otkk', fornavn: 'Synne', etternavn: 'Lunde' }),
            journalfortAv: { ident: '3t55yt', fornavn: 'Victoria', etternavn: 'Andresen' },
            fritekst:
                'Animi qui molestiae omnis reiciendis quam in. Provident magnam dolor. Omnis voluptatem placeat labore in ut aliquam perspiciatis fugiat. Perspiciatis voluptatum odit.',
            lestDato: '2019-08-11',
            status: LestStatus.Lest,
            opprettetDato: '2019-08-06',
            journalfortDato: '2019-08-12',
            ferdigstiltDato: '2019-07-12',
            erFerdigstiltUtenSvar: true,
            erDokumentMelding: false
        },
        {
            id: '90l2nkx0',
            meldingstype: Meldingstype.DELVIS_SVAR_SKRIFTLIG,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'xy70r8', fornavn: 'Kristoffer', etternavn: 'Fredriksen' }),
            journalfortAv: { ident: 'p0jh0d', fornavn: 'Jenny', etternavn: 'Berg' },
            fritekst:
                'Dolores omnis ea voluptatem corrupti odit incidunt enim. Unde fugiat sed animi explicabo maiores. Nihil cum accusamus omnis officiis deserunt est. Qui quisquam unde aut magni dignissimos.',
            lestDato: '2019-07-23',
            status: LestStatus.Lest,
            opprettetDato: '2019-08-13',
            journalfortDato: '2019-07-30',
            ferdigstiltDato: '2019-07-27',
            erFerdigstiltUtenSvar: false,
            erDokumentMelding: false
        }
    ]
};
