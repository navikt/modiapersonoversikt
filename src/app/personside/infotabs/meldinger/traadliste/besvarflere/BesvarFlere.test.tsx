import * as React from 'react';
import TestProvider from '../../../../../../test/Testprovider';
import BesvarFlere from './BesvarFlere';
import { statiskTraadMock } from '../../../../../../mock/meldinger/statiskTraadMock';
import * as renderer from 'react-test-renderer';

test('BesvarFler matcher snapshot', () => {
    const besvarFler = renderer.create(
        <TestProvider>
            <BesvarFlere traader={[statiskTraadMock, statiskTraadMock]} lukkModal={() => null} />
        </TestProvider>
    );

    expect(besvarFler.toJSON()).toMatchSnapshot();
});
