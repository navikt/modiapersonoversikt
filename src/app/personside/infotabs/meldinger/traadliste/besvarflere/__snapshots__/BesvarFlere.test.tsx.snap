// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BesvarFler matcher snapshot 1`] = `
.c9 {
  position: relative;
}

.c9 svg {
  height: 1.5625rem;
  width: 1.5625rem;
  opacity: 0.4;
}

.c10 {
  position: absolute;
  top: -0.1875rem;
  right: -0.5rem;
  background-color: #78706a;
  color: white;
  height: 1.1rem;
  width: 1.1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  width: 0;
  margin-left: 0.8rem;
  overflow-wrap: break-word;
}

.c11 .order-first {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c11 .order-second {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c11 .order-third {
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-top: 0.2rem;
}

.c13 > *:not(:last-child) {
  margin-bottom: 0.2rem;
  margin-right: 0.2rem;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 .order-first {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c12 .order-second {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 > *:last-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c14 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  padding: .5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  background-color: #cce1f3;
}

.c4:focus-within {
  position: relative;
  outline: none;
  box-shadow: none;
}

.c4:focus-within::after {
  position: absolute;
  top: 0;
  left: 0;
  content: '';
  box-shadow: inset 0 0 0 0.1875rem #254b6d;
  height: 100%;
  width: 100%;
  pointer-events: none;
}

.c4 .hover-animation {
  -webkit-transition: -webkit-transform 0.3s;
  -webkit-transition: transform 0.3s;
  transition: transform 0.3s;
}

.c4:hover {
  box-shadow: inset 0 0 0 0.1rem #0067c5,0 0.1rem 0.1rem rgba(0,0,0,0.2);
}

.c4:hover .hover-animation {
  -webkit-transform: translateX(0.5rem);
  -ms-transform: translateX(0.5rem);
  transform: translateX(0.5rem);
}

.c16:focus-within {
  position: relative;
  outline: none;
  box-shadow: none;
}

.c16:focus-within::after {
  position: absolute;
  top: 0;
  left: 0;
  content: '';
  box-shadow: inset 0 0 0 0.1875rem #254b6d;
  height: 100%;
  width: 100%;
  pointer-events: none;
}

.c16 .hover-animation {
  -webkit-transition: -webkit-transform 0.3s;
  -webkit-transition: transform 0.3s;
  transition: transform 0.3s;
}

.c16:hover {
  box-shadow: inset 0 0 0 0.1rem #0067c5,0 0.1rem 0.1rem rgba(0,0,0,0.2);
}

.c16:hover .hover-animation {
  -webkit-transform: translateX(0.5rem);
  -ms-transform: translateX(0.5rem);
  transform: translateX(0.5rem);
}

.c15 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c15 .nav-frontend-chevron {
  width: 1.5625rem;
}

.c18 .snakkeboble-panel {
  -webkit-flex-basis: 40rem;
  -ms-flex-preferred-size: 40rem;
  flex-basis: 40rem;
}

.c18 a {
  word-break: break-word;
}

.c18 .snakkeboble {
  margin: 0;
}

.c21 {
  font-weight: 600;
}

.c19 {
  text-align: left;
}

.c19 em {
  font-style: normal;
  background-color: #eed28c;
  box-shadow: 0 0 0 0.2rem #eed28c;
  border-radius: 0.625rem;
}

.c19 > *:not(:last-child) {
  border-bottom: solid 0.0625rem #b7b1a9;
  padding-bottom: 0.7rem;
  margin-bottom: 0.5rem;
}

.c24 {
  color: #0067c5;
}

.c23 .ekspanderbartPanel__hode,
.c23 .ekspanderbartPanel__innhold {
  padding: 0.3rem 0;
}

.c22 p {
  margin-bottom: 0 !important;
}

.c0 {
  background-color: #c6c2bf;
  width: 80vw;
  height: 80vh;
  max-width: 100rem;
  max-height: 100rem;
  display: -ms-grid;
  display: grid;
  -ms-grid-rows: auto minmax(0,1fr) auto;
  grid-template-rows: auto minmax(0,1fr) auto;
  -ms-grid-columns: 1fr;
  grid-template-columns: 1fr;
}

.c0 > *:nth-child(1) {
  -ms-grid-row: 1;
}

.c0 > *:nth-child(2) {
  -ms-grid-row: 2;
}

.c0 > *:nth-child(3) {
  -ms-grid-row: 3;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 > *:first-child {
  -webkit-flex: 35% 1 1;
  -ms-flex: 35% 1 1;
  flex: 35% 1 1;
}

.c2 > *:last-child {
  -webkit-flex: 65% 1 1;
  -ms-flex: 65% 1 1;
  flex: 65% 1 1;
}

.c2 > * {
  overflow: auto;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: white;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  list-style: none;
}

.c3 > * {
  border-bottom: solid 0.0625rem #b7b1a9;
}

.c17 {
  padding: .5rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c6 {
  margin: 0 0.5rem;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-transform: translateY(-0.5rem);
  -ms-transform: translateY(-0.5rem);
  transform: translateY(-0.5rem);
}

.c1 {
  background-color: #e9e7e7;
  padding: 1.25rem .5rem;
}

.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  background-color: #e9e7e7;
  padding: .5rem;
}

.c25 > * {
  margin-left: 1rem;
}

@media print {
  .c18 {
    page-break-inside: avoid;
    border: solid 0.0625rem #78706a;
    margin-bottom: 2rem;
  }

  .c18 .snakkeboble__snakkeboble-pil-container,
  .c18 .nav-ikon,
  .c18 .bruker-ikon {
    display: none;
  }

  .c18 .snakkeboble-panel {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
  }
}

@media print {
  .c23 {
    display: none;
  }
}

<form
  className="c0"
  onSubmit={[Function]}
>
  <div
    className="c1"
  >
    <p
      className="typo-ingress"
    >
      Slå sammen oppgaver
    </p>
  </div>
  <div
    className="c2"
  >
    <ol
      className="c3"
      role="tablist"
    >
      <li
        className="c4"
      >
        <input
          checked={true}
          className="sr-only test-meldinger-melding valgt_melding  "
          id="Helt tilfeldig ID"
          name="traadliste-slaa-sammen"
          onChange={[Function]}
          type="radio"
          value="sg838exr"
        />
        <label
          className="c5"
          htmlFor="Helt tilfeldig ID"
        >
          <div
            className="c6"
          >
            <div
              className="skjemaelement"
            >
              <input
                aria-invalid={false}
                aria-label="Velg tråd"
                checked={false}
                className="skjemaelement__input checkboks"
                id="Helt tilfeldig ID"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                
              </label>
            </div>
          </div>
          <div
            className="c7"
          >
            <div
              className="c8"
            >
              <span
                className="sr-only"
              >
                (
                4
                )
              </span>
              <span
                className="c9"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="#3e3832"
                  >
                    <path
                      d="M16.5 8.75c.476 0 .941.045 1.399.115.065-.379.101-.771.101-1.18C18 3.585 13.962.25 9 .25c-4.963 0-9 3.335-9 7.435 0 1.962.855 3.826 2.36 5.178l-1.794 3.14a.5.5 0 00.62.712l4.839-1.937c.435.124.887.218 1.348.281.022.002.065-.005.075.004.075 0 .319-.002.678-.022.679-3.514 4.233-6.291 8.374-6.291z"
                    />
                    <path
                      d="M16.5 9.75c-4.065 0-7.5 2.977-7.5 6.5 0 3.355 2.878 6.224 6.434 6.572a7.581 7.581 0 003.578-.534l3.813 1.43a.498.498 0 00.623-.691l-1.325-2.65C23.327 19.289 24 17.868 24 16.25c0-3.523-3.435-6.5-7.5-6.5z"
                    />
                  </g>
                </svg>
                <p
                  aria-hidden="true"
                  className="typo-undertekst c10"
                >
                  4
                </p>
              </span>
              <div
                className="c11"
              >
                <div
                  className="c12 order-first"
                >
                  <p
                    className="typo-element order-second"
                  >
                    Delsvar - Arbeid
                  </p>
                  <p
                    className="typo-normal order-first"
                  >
                    27.07.2019 00:00
                  </p>
                </div>
                <div
                  className="c13 order-third"
                >
                  <div
                    className="etikett etikett--fokus"
                  >
                    <span
                      className="typo-normal"
                    >
                      Under arbeid
                    </span>
                  </div>
                  <div
                    className="etikett etikett--info"
                  >
                    <span
                      className="typo-normal"
                    >
                      Delvis besvart
                    </span>
                  </div>
                </div>
                <p
                  className="typo-normal c14 order-second"
                >
                  Dolores omnis ea voluptatem corrupti odit incidunt enim. Unde fugiat sed animi explicabo maiores. Nihil cum accusamus omnis officiis deserunt est. Qui quisquam unde aut magni dignissimos.
                </p>
              </div>
            </div>
          </div>
          <div
            className="c15 hover-animation"
          >
            <i
              className="nav-frontend-chevron chevronboks chevron--hoyre"
            />
          </div>
        </label>
      </li>
      <li
        className="c16"
      >
        <input
          checked={false}
          className="sr-only test-meldinger-melding   "
          id="Helt tilfeldig ID"
          name="traadliste-slaa-sammen"
          onChange={[Function]}
          type="radio"
          value="123"
        />
        <label
          className="c5"
          htmlFor="Helt tilfeldig ID"
        >
          <div
            className="c6"
          >
            <div
              className="skjemaelement"
            >
              <input
                aria-invalid={false}
                aria-label="Velg tråd"
                checked={false}
                className="skjemaelement__input checkboks"
                id="Helt tilfeldig ID"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                
              </label>
            </div>
          </div>
          <div
            className="c7"
          >
            <div
              className="c8"
            >
              <span
                className="sr-only"
              >
                (
                4
                )
              </span>
              <span
                className="c9"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="#3e3832"
                  >
                    <path
                      d="M16.5 8.75c.476 0 .941.045 1.399.115.065-.379.101-.771.101-1.18C18 3.585 13.962.25 9 .25c-4.963 0-9 3.335-9 7.435 0 1.962.855 3.826 2.36 5.178l-1.794 3.14a.5.5 0 00.62.712l4.839-1.937c.435.124.887.218 1.348.281.022.002.065-.005.075.004.075 0 .319-.002.678-.022.679-3.514 4.233-6.291 8.374-6.291z"
                    />
                    <path
                      d="M16.5 9.75c-4.065 0-7.5 2.977-7.5 6.5 0 3.355 2.878 6.224 6.434 6.572a7.581 7.581 0 003.578-.534l3.813 1.43a.498.498 0 00.623-.691l-1.325-2.65C23.327 19.289 24 17.868 24 16.25c0-3.523-3.435-6.5-7.5-6.5z"
                    />
                  </g>
                </svg>
                <p
                  aria-hidden="true"
                  className="typo-undertekst c10"
                >
                  4
                </p>
              </span>
              <div
                className="c11"
              >
                <div
                  className="c12 order-first"
                >
                  <p
                    className="typo-element order-second"
                  >
                    Delsvar - Arbeid
                  </p>
                  <p
                    className="typo-normal order-first"
                  >
                    27.07.2019 00:00
                  </p>
                </div>
                <div
                  className="c13 order-third"
                >
                  <div
                    className="etikett etikett--fokus"
                  >
                    <span
                      className="typo-normal"
                    >
                      Under arbeid
                    </span>
                  </div>
                  <div
                    className="etikett etikett--info"
                  >
                    <span
                      className="typo-normal"
                    >
                      Delvis besvart
                    </span>
                  </div>
                </div>
                <p
                  className="typo-normal c14 order-second"
                >
                  Dolores omnis ea voluptatem corrupti odit incidunt enim. Unde fugiat sed animi explicabo maiores. Nihil cum accusamus omnis officiis deserunt est. Qui quisquam unde aut magni dignissimos.
                </p>
              </div>
            </div>
          </div>
          <div
            className="c15 hover-animation"
          >
            <i
              className="nav-frontend-chevron chevronboks chevron--hoyre"
            />
          </div>
        </label>
      </li>
    </ol>
    <ol
      className="c17"
      role="tabpanel"
    >
      <li
        className="c18 snakkeboble_ikoner"
      >
        <article
          aria-labelledby="Helt tilfeldig ID"
        >
          <div
            className="snakkeboble snakkeboble--hoyre"
          >
            <i
              className="nav-ikon"
            />
            <div
              className="snakkeboble__snakkeboble-pil-container"
            >
              <i
                className="snakkeboble__snakkebole-pil"
              />
            </div>
            <div
              className="panel snakkeboble-panel"
            >
              <div
                className="snakkeboble-panel__tekst"
              >
                <div
                  className="c19"
                >
                  <div
                    className=""
                  >
                    <div
                      className="c20"
                    >
                      <h4
                        id="Helt tilfeldig ID"
                      >
                        <span
                          className="sr-only"
                        >
                          Melding 
                          4
                        </span>
                        <div
                          className="c21"
                        >
                          Samtalereferat oppmøte - Arbeid
                        </div>
                      </h4>
                      <div
                        className="etikett etikett--suksess"
                      >
                        <span
                          className="typo-normal"
                        >
                          Lest
                        </span>
                      </div>
                    </div>
                    <div>
                      07.07.2019 00:00
                    </div>
                    <div
                      className="c22 typo-normal"
                    >
                      Skrevet av Sander Hansen (nzxo4q)
                    </div>
                  </div>
                  <div>
                    <p
                      className="typo-normal"
                    >
                      Ea laborum accusamus et ex voluptates provident eveniet voluptatem. Sunt et sed eligendi harum exercitationem. Eaque est maiores autem sunt occaecati atque ipsa ea itaque. Dolorum enim quaerat alias ut sit labore pariatur.
                    </p>
                  </div>
                  <div
                    className="ekspanderbartPanel c23 ekspanderbartPanel--lukket"
                  >
                    <button
                      aria-controls="Helt tilfeldig ID"
                      aria-expanded={false}
                      className="ekspanderbartPanel__hode"
                      id="Helt tilfeldig ID"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      type="button"
                    >
                      <div
                        className="ekspanderbartPanel__flex-wrapper"
                      >
                        <span
                          className="ekspanderbartPanel__tittel"
                        >
                          <p
                            className="typo-etikett-liten c24"
                          >
                            Meldingen er journalført
                          </p>
                        </span>
                        <span
                          className="ekspanderbartPanel__indikator"
                        />
                      </div>
                    </button>
                    <div
                      aria-labelledby="Helt tilfeldig ID"
                      id="Helt tilfeldig ID"
                      role="region"
                    >
                      <div
                        className="ekspanderbartPanel__innhold"
                      >
                        Journalført av Adrian Pettersen (vlzaiw) 11.07.2019 på ukjent tema med ukjent saksid
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </article>
      </li>
      <li
        className="c18 snakkeboble_ikoner"
      >
        <article
          aria-labelledby="Helt tilfeldig ID"
        >
          <div
            className="snakkeboble snakkeboble--venstre"
          >
            <i
              className="bruker-ikon"
            />
            <div
              className="snakkeboble__snakkeboble-pil-container"
            >
              <i
                className="snakkeboble__snakkebole-pil"
              />
            </div>
            <div
              className="panel snakkeboble-panel"
            >
              <div
                className="snakkeboble-panel__tekst"
              >
                <div
                  className="c19"
                >
                  <div
                    className=""
                  >
                    <div
                      className="c20"
                    >
                      <h4
                        id="Helt tilfeldig ID"
                      >
                        <span
                          className="sr-only"
                        >
                          Melding 
                          3
                        </span>
                        <div
                          className="c21"
                        >
                          Spørsmål fra bruker - Arbeid
                        </div>
                      </h4>
                    </div>
                    <div>
                      16.07.2019 00:00
                    </div>
                  </div>
                  <div>
                    <p
                      className="typo-normal"
                    >
                      Quo magni eaque et libero minima ipsum amet et dolorem. Sed iusto ratione sit. Voluptate alias qui impedit ut dolorum doloribus qui. Qui beatae rerum minima suscipit.
                    </p>
                  </div>
                  <div
                    className="ekspanderbartPanel c23 ekspanderbartPanel--lukket"
                  >
                    <button
                      aria-controls="Helt tilfeldig ID"
                      aria-expanded={false}
                      className="ekspanderbartPanel__hode"
                      id="Helt tilfeldig ID"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      type="button"
                    >
                      <div
                        className="ekspanderbartPanel__flex-wrapper"
                      >
                        <span
                          className="ekspanderbartPanel__tittel"
                        >
                          <p
                            className="typo-etikett-liten c24"
                          >
                            Meldingen er journalført
                          </p>
                        </span>
                        <span
                          className="ekspanderbartPanel__indikator"
                        />
                      </div>
                    </button>
                    <div
                      aria-labelledby="Helt tilfeldig ID"
                      id="Helt tilfeldig ID"
                      role="region"
                    >
                      <div
                        className="ekspanderbartPanel__innhold"
                      >
                        Journalført av Emil Nguyen (ly4pps) 21.07.2019 på ukjent tema med ukjent saksid
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </article>
      </li>
      <li
        className="c18 snakkeboble_ikoner"
      >
        <article
          aria-labelledby="Helt tilfeldig ID"
        >
          <div
            className="snakkeboble snakkeboble--venstre"
          >
            <i
              className="bruker-ikon"
            />
            <div
              className="snakkeboble__snakkeboble-pil-container"
            >
              <i
                className="snakkeboble__snakkebole-pil"
              />
            </div>
            <div
              className="panel snakkeboble-panel"
            >
              <div
                className="snakkeboble-panel__tekst"
              >
                <div
                  className="c19"
                >
                  <div
                    className=""
                  >
                    <div
                      className="c20"
                    >
                      <h4
                        id="Helt tilfeldig ID"
                      >
                        <span
                          className="sr-only"
                        >
                          Melding 
                          2
                        </span>
                        <div
                          className="c21"
                        >
                          Spørsmål fra bruker - Arbeid
                        </div>
                      </h4>
                    </div>
                    <div>
                      12.07.2019 00:00
                    </div>
                  </div>
                  <div>
                    <p
                      className="typo-normal"
                    >
                      Animi qui molestiae omnis reiciendis quam in. Provident magnam dolor. Omnis voluptatem placeat labore in ut aliquam perspiciatis fugiat. Perspiciatis voluptatum odit.
                    </p>
                  </div>
                  <div
                    className="ekspanderbartPanel c23 ekspanderbartPanel--lukket"
                  >
                    <button
                      aria-controls="Helt tilfeldig ID"
                      aria-expanded={false}
                      className="ekspanderbartPanel__hode"
                      id="Helt tilfeldig ID"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      type="button"
                    >
                      <div
                        className="ekspanderbartPanel__flex-wrapper"
                      >
                        <span
                          className="ekspanderbartPanel__tittel"
                        >
                          <p
                            className="typo-etikett-liten c24"
                          >
                            Meldingen er journalført
                          </p>
                        </span>
                        <span
                          className="ekspanderbartPanel__indikator"
                        />
                      </div>
                    </button>
                    <div
                      aria-labelledby="Helt tilfeldig ID"
                      id="Helt tilfeldig ID"
                      role="region"
                    >
                      <div
                        className="ekspanderbartPanel__innhold"
                      >
                        Journalført av Victoria Andresen (3t55yt) 12.08.2019 på ukjent tema med ukjent saksid
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </article>
      </li>
      <li
        className="c18 snakkeboble_ikoner"
      >
        <article
          aria-labelledby="Helt tilfeldig ID"
        >
          <div
            className="snakkeboble snakkeboble--hoyre"
          >
            <i
              className="nav-ikon"
            />
            <div
              className="snakkeboble__snakkeboble-pil-container"
            >
              <i
                className="snakkeboble__snakkebole-pil"
              />
            </div>
            <div
              className="panel snakkeboble-panel"
            >
              <div
                className="snakkeboble-panel__tekst"
              >
                <div
                  className="c19"
                >
                  <div
                    className=""
                  >
                    <div
                      className="c20"
                    >
                      <h4
                        id="Helt tilfeldig ID"
                      >
                        <span
                          className="sr-only"
                        >
                          Melding 
                          1
                        </span>
                        <div
                          className="c21"
                        >
                          Delsvar - Arbeid
                        </div>
                      </h4>
                      <div
                        className="etikett etikett--info"
                      >
                        <span
                          className="typo-normal"
                        >
                          Delsvar
                        </span>
                      </div>
                    </div>
                    <div>
                      27.07.2019 00:00
                    </div>
                    <div
                      className="c22 typo-normal"
                    >
                      Skrevet av Kristoffer Fredriksen (xy70r8)
                    </div>
                  </div>
                  <div>
                    <p
                      className="typo-normal"
                    >
                      Dolores omnis ea voluptatem corrupti odit incidunt enim. Unde fugiat sed animi explicabo maiores. Nihil cum accusamus omnis officiis deserunt est. Qui quisquam unde aut magni dignissimos.
                    </p>
                  </div>
                  <div
                    className="ekspanderbartPanel c23 ekspanderbartPanel--lukket"
                  >
                    <button
                      aria-controls="Helt tilfeldig ID"
                      aria-expanded={false}
                      className="ekspanderbartPanel__hode"
                      id="Helt tilfeldig ID"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      type="button"
                    >
                      <div
                        className="ekspanderbartPanel__flex-wrapper"
                      >
                        <span
                          className="ekspanderbartPanel__tittel"
                        >
                          <p
                            className="typo-etikett-liten c24"
                          >
                            Meldingen er journalført
                          </p>
                        </span>
                        <span
                          className="ekspanderbartPanel__indikator"
                        />
                      </div>
                    </button>
                    <div
                      aria-labelledby="Helt tilfeldig ID"
                      id="Helt tilfeldig ID"
                      role="region"
                    >
                      <div
                        className="ekspanderbartPanel__innhold"
                      >
                        Journalført av Jenny Berg (p0jh0d) 30.07.2019 på ukjent tema med ukjent saksid
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </article>
      </li>
    </ol>
  </div>
  <div
    className="c25"
  >
    <button
      className="knapp knapp--hoved"
      disabled={false}
      type="submit"
    >
      Slå sammen 
      oppgaver
    </button>
  </div>
</form>
`;
